packages/components/src/FormField/FormField.tsx

```diff
@@ -1,426 +1,216 @@
-
-  inputRef?: RefObject<
-    HTMLInputElement | HTMLTextAreaElement | HTMLSelectElement
-  >;
-}
-
-/**
- * Max statements is disabled for this component due the event handlers.
- */
-// eslint-disable-next-line max-statements
-export function FormField({
-  align,
-  autocomplete = true,
-  children,
-  defaultValue,
-  disabled,
-  inline,
-  invalid,
-  max,
-  maxLength,
-  min,
-  name,
-  onFocus,
-  onBlur,
-  onChange,
-  onEnter,
-  onValidation,
-  loading,
-  placeholder,
-  readonly,
-  rows,
-  size,
-  type = "text",
-  keyboard,
-  value,
-  validations,
-  inputRef,
-}: FormFieldProps) {
+export function FormField(props: FormFieldProps) {
   const {
-    register,
-    setValue,
-    control,
-    /**
-     * We currently only use `errors` from formState, but there are lots of
-     * other values that come with it. https://react-hook-form.com/api#formState
-     */
-    formState: { errors },
-  } =
+    actionsRef,
+    autocomplete = true,
+    children,
+    defaultValue,
+    disabled,
+    inline,
+    inputRef,
+    keyboard,
+    loading,
+    max,
+    maxLength,
+    min,
+    name,
+    readonly,
+    rows,
+    type = "text",
+    validations,
+    value,
+    onChange,
+    onEnter,
+    onFocus,
+    onBlur,
+    onValidation,
+  } = props;
+
+  const { control, errors, setValue } =
     useFormContext() != undefined
       ? useFormContext()
       : useForm({ mode: "onTouched" });
 
-  const [hasMiniLabel, setHasMiniLabel] = useState(
-    shouldShowMiniLabel(defaultValue, value),
-  );
   const [identifier] = useState(uuid.v1());
-  const isControlled = value !== undefined;
 
-  if (!name && (validations || isControlled)) {
-    name = `generatedName--${identifier}`;
-  }
-
-  const error = name && errors[name] && errors[name].message;
+  /**
+   * Generate a name if one is not supplied, this is the name
+   * that will be used for react-hook-form and not neccessarily
+   * attached to the DOM
+   */
+  const [controlledName] = useState(
+    name ? name : `generatedName--${identifier}`,
+  );
 
-  useEffect(() => setHasMiniLabel(shouldShowMiniLabel(defaultValue, value)), [
-    value,
-  ]);
-  useEffect(() => handleValidation(), [error]);
+  const [hasMiniLabel, setHasMiniLabel] = useState(
+    shouldShowMiniLabel(defaultValue, value),
+  );
 
-  const autocompleteValue = setAutocomplete(autocomplete);
+  useEffect(() => {
+    if (value != undefined) {
+      setValue(controlledName, value);
+      setHasMiniLabel(String(value).length > 0);
+    }
+  }, [value]);
 
-  const wrapperClassNames = classnames(
-    styles.wrapper,
-    inline && styles.inline,
-    size && styles[size],
-    align && styles[align],
-    disabled && styles.disabled,
-    maxLength && styles.maxLength,
-    {
-      [styles.miniLabel]:
-        (hasMiniLabel || type === "time" || type === "select") && placeholder,
-      [styles.invalid]: invalid || error,
+  useImperativeHandle(actionsRef, () => ({
+    setValue: newValue => {
```

Comment:
Now that we have the `useEffect` above this I'm pondering if we still need this. I feel like the answer is we do but I'm not 100% sure… 🤔

Reply:
I think that we do as this still fixes the insertion issue where the cursor goes to the end of the input.

Reply:
👍